nixos/modules/services/web-apps/ackee.nix

```diff
@@ -0,0 +1,135 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.ackee;
+  settings = cfg.defaultSettings // cfg.settings;
+in
+{
+  options.services.ackee = {
+    enable = mkEnableOption "the Ackee analytics tool";
+
+    defaultSettings = mkOption {
+      type = with types; attrsOf (nullOr (oneOf [ str path package ]));
+      default = {
+        ACKEE_MONGODB = "mongodb://localhost:27017/ackee";
+        ACKEE_USERNAME = "ackee";
+        ACKEE_PORT = "3000";
+      };
+      example = literalExample "{}";
+      description = ''
+        Settings can be overwritten in <option>settings</option>.
+        To remove all default settings, set <code>defaultSettings = {}</code>.
+      '';
+    };
+
+    settings = mkOption {
+      type = with types; attrsOf (nullOr (oneOf [ str path package ]));
+      default = { };
+      example = literalExample ''
+        {
+          ACKEE_MONGODB = "mongodb://localhost:27017/ackee";
+          ACKEE_USERNAME = "ackee";
+          ACKEE_PORT = "3000";
+        }
+      '';
+      description = ''
+        The configuration of Ackee is done through environment variables. The variables defined in
+        this settings will be set on the systemd service.
+
+        The available options can be found in
+        <link xlink:href="https://github.com/electerious/Ackee/blob/v${pkgs.ackee.version}/docs/Options.md">the documentation</link>.
+      '';
+    };
+
+    settingsFile = mkOption {
```

Comment:
Are there multiple credentials that can be provided to the application, or just the one? I think you should consider renaming this option to `environmentFile`.

Reply:
`ACKEE_MONGODB` can also contain credentials if the mongodb has a password, so I wanted to leave that option open.

Reply:
Sounds good. I still suggest renaming to `environmentFile`.